src/core/unit.c

```diff
@@ -4695,15 +4679,12 @@ int unit_setup_exec_runtime(Unit *u) {
 
         /* Try to get it from somebody else */
         HASHMAP_FOREACH_KEY(v, other, u->dependencies[UNIT_JOINS_NAMESPACE_OF], i) {
-
-                *rt = unit_get_exec_runtime(other);
-                if (*rt) {
-                        exec_runtime_ref(*rt);
-                        return 0;
-                }
+                r = exec_runtime_acquire(u->manager, NULL, other->id, false, rt);
+                if (r == 1)
+                        return 1;
```

Comment:
I think that this needs a log line in case `r < 0`. This shouldn't happen, but if it does, `exec_runtime_acquire` does not do any logging on its own, so we wouldn't ever know.

Reply:
Yeah, that is true in general. But the function returns only 0 or 1 when the fourth parameter is false...

Reply:
OK. It seems all the callers log the failure at some point.